### ISSUE TITLE
Batch Normalization in Recurrent Networks

### ISSUE THREAD

**Issue by duguyue100** [fchollet/keras#5637]
Hi,

I'm trying to implement batch normalisation in recurrent networks.
Instead of writing another batch normalisation in the class, I would like to find a way of using `BatchNormalization` layer. But it doesn't seem to work.
__Is there anyway of doing something like this?__

Here is what I did, I init one `BatchNormalization` layer in the `__init__` function and build it in my `build` function. Since it's in the model, I added the `trainable_weights` and `non_trainable_weights` in the layer's weight's list. And finally I call it in the `call` function but then the model couldn't compile.

The conclusion is __I think I did something wrong and nasty, so can someone give a small example that works?__

Thanks in advance!

**Comment by brortao** [comment]
Recurrent batch normalization is a bit different from standard batch normalization. See [arxiv/1603.09025](https://www.arxiv.org/abs/1603.09025). AFAIK it's not implemented in Keras yet, but I'd love to be proved wrong!

**Comment by duguyue100** [comment]
@brortao Thanks! I will read the paper again.
